### ISSUE TITLE
Can't choose session (Xephyr)

### ISSUE THREAD

**Issue by SolarAquarion** [Obsidian-StudiosInc/entrance#16]
[deepinscreenshot_select-area_20180423120758](https://user-images.githubusercontent.com/5307016/39139093-10644abc-46ef-11e8-9d43-77bd1d2198cf.png)
This is also true for when i do it via systemd

**Comment by wltjr** [comment]
Not sure I understand, your running this in Xephyr? Is this what you were talking about in issue #12? There is another bug there I need to fix when stopping Xephyr, client will restart.

Seems a few things are going wrong, your missing icons for your user, seems like the system actions are missing in the bottom left corner. Though seems like you are getting sessions LXDE? That seems to be populating. I would assume its empty like the others.

Maybe something is failing there, bad session desktop file or failing to parse. I have seen that before when I first messed with Entrance. That happens when the client fails to connect to server or has communication issues. It should all be empty, including where it shows LXDE.

I bet you have an efreetd, or runtime issue. Can you provide a log file? I bet you have something going wrong, and its failing to create a directory or something for a socket. I had lots of issues with such under Travis recently.

Also I assume your using the latest from git? EFL 1.20.7? I need to do a new release soon, just keep doing so much. Then again I can keep releasing...

**Comment by SolarAquarion** [comment]
that is what i was describing with issue #12 that i couldn't choose
anything but LXDE or similar.

**Comment by wltjr** [comment]
Please provide a copy of your entrance log file, _/var/log/entrance.log_. I am pretty sure something is going wrong. Either [efreetd cannot create a directory](https://travis-ci.org/Obsidian-StudiosInc/entrance/jobs/362959773#L1203) related to XDG_RUNTIME_DIR and/or issues with the [socket file under /tmp](https://travis-ci.org/Obsidian-StudiosInc/entrance/jobs/365893507#L1263). Both those are user permission issues, and likely one or the other is the cause of  your problems. For sure the 2nd. Or something else is showing in log.

The client is having communication issues with the server. Your missing more than that. Seems you have no restart, reboot, shutdown options in the bottom left corner. That should open up to a menu as well. That you have LXDE is odd, seems its able to transmit that and fails after.

**Comment by SolarAquarion** [comment]
[entrance.log](https://github.com/Obsidian-StudiosInc/entrance/files/1939309/entrance.log)

What do you see in my log?

**Comment by wltjr** [comment]
Lets start with this issue
```
ERR<808>:elementary lib/elementary/elm_layout.c:893 _elm_layout_efl_file_file_set() failed to set edje file '/usr/share/entrance/themes/default.edj', group 'entrance/background': Unknown Collection
```

Can you verify _/usr/share/entrance/themes/default.edj_ exists and is publicly readable? Seems that maybe missing. If so that is likely the problem. The type of fields is set in edj not code. So it has no clue what type to fill that stuff with. It supports 3 types.

```
INF<764>:entrance_client src/daemon/entrance.c:191 _entrance_client_data() src/bin/entrance_connect.c:87 Users received
src/bin/entrance_gui.c:450 Add users list, using item style: default
src/bin/entrance_fill.c:227 Unknow object type to fill Elm_Gengrid

INF<764>:entrance_client src/daemon/entrance.c:191 _entrance_client_data() src/bin/entrance_connect.c:92 Action received
src/bin/entrance_gui.c:408 Actions set
src/bin/entrance_fill.c:227 Unknow object type to fill Elm_Hoversel

INF<764>:entrance_client src/daemon/entrance.c:191 _entrance_client_data() src/bin/entrance_connect.c:82 Xsession received
src/bin/entrance_login.c:557 Session set
src/bin/entrance_fill.c:227 Unknow object type to fill Elm_Hoversel
```

It is odd that any sessions show up since no users or actions show up. Both actions and sessions are Hoversel/selectboxes. Its one thing I likely will leave as it allows different widgets to be used in different themes. I am using hoversel, but it could use other types, gengrid, genlist or list. Thus where it says unknown, it doesn't know what type, hoversel, gengrid, genlist, or list. Not without a edj file to tell it which to use.

**Comment by SolarAquarion** [comment]
ls /usr/share/entrance/themes/default.edj it exists
-rw-r--r-- 1 root root 1331897 Apr 23 13:30
/usr/share/entrance/themes/default.edj

**Comment by SolarAquarion** [comment]
![deepinscreenshot_select-area_20180423133407](https://user-images.githubusercontent.com/5307016/39143173-0c6f2c72-46fb-11e8-92ec-a37b488b47a0.png)

Got a new photo

**Comment by wltjr** [comment]
Git maybe less of an issue with entrance. EFL git causes issues for ecrire. For ecrire it is best to not use git. I need to modify ecrire build systems for such. There are [issues with ecrire](https://github.com/Obsidian-StudiosInc/ecrire/issues/29#issuecomment-352875398) under git. These entrance issues do not seem git related.

Your edj file exists and permissions are fine... Of course it would not be that easy.

In that screenshot I see you have actions now in the bottom left. Seems you may have sessions now as well. Can you select any? One will not be selected by default if you had not previously logged in, one saved in history. 

Your missing background is odd. This is strange ```group 'entrance/background': Unknown Collection```. Looking into that, and not sure where that is coming from. That is not in use anywhere. Are you setting a background file in your _entrance.conf_? I only tested images. Something maybe going wrong there.

**Comment by wltjr** [comment]
By the way I just pushed an update to fix running under Xephyr. The client will restart on close and take over your desktop, or a display if more than one.

**Comment by wltjr** [comment]
I would try this under EFL 1.20.7. You have some odd output. The part of the one exception I keep looking at is very odd. This file was removed from git.

```
ERR<808>:elementary lib/elementary/elm_layout.c:893
```

The file exists in [1.20.7](https://github.com/Enlightenment/efl/blob/v1.20.7/src/lib/elementary/elm_layout.c), but not [git master](https://github.com/Enlightenment/efl/blob/master/src/lib/elementary/elm_layout.c). The [header still exists](https://github.com/Enlightenment/efl/blob/master/src/lib/elementary/elm_layout.h) in master, maybe converted to EO or something.

Maybe you need to update your git EFL?

**Comment by SolarAquarion** [comment]
I can't select any will delete the .log and create a new one

**Comment by wltjr** [comment]
Ok thanks! This is pretty strange. I have yet to see these issues. Maybe due to the amount of sessions or a bad desktop file. That entrance is failing to parse and causing other issues. But that edje issue with the background is odd. You had a background in your first shot, and not in your 2nd. But the group it was looking for does not exist. No clue where that is coming from, but for sure will fail as its a non existent group.

**Comment by wltjr** [comment]
Or you can try to move some of those session files out, reduce them all to like 1 or 2 and see if problem remains. That you can try locally and will rule out the amount of sessions or a bad desktop file format/syntax.

**Comment by SolarAquarion** [comment]
Well, the problem was a mix of the log and the efl-git

**Comment by wltjr** [comment]
Are things working now? At least as good as Entrance works in general? Though should be fairly reliable now, that I fixed most all stopping issues. Short of when X hangs or something goes wrong.

**Comment by SolarAquarion** [comment]
it works now although the avatars are weird

**Comment by wltjr** [comment]
I am not sure if I will keep those. Not sure if there is a need for images. They just went with some tux ones I forget from where. They are some what common. I think you are supposed to be able to configure your own in entrance.conf file. I have not tried that.

At least you have sessions now. I think all should be good there. I added back processing like xinitrc.d, and other stuff. Though I get problems on Gentoo from Xresources. Seems to be Gentoo specific.

Anyway feel free to close if this is resolved, and open any issues for anything you run into. I have a few things to dial in for X. Then I will switch over to Wayland stuff. Moving hard coded stuff to conf file, like display :0, and also vt, since that is set as an env var. People may want to use a vt other than 7, and display other than :0.0. Which I think :0.0 vs :0 caused me some problems on dual display.

**Comment by wltjr** [comment]
Once I get the code furthered I will play with the look and theme more and maybe some other themes. I liked a [mockup](https://github.com/wfx/enter/blob/master/data/mock.svg) they made for lightdm python EFL interface [enter](https://github.com/wfx/enter).

**Comment by wltjr** [comment]
Yeah I may play with a new theme like that. That stuff will come later one. I also want to animate the current one a bit more. Have something animated in background etc. E logo spinning or something.
